src/stan/lang/generator/generate_constructor.hpp

```diff
@@ -26,27 +26,28 @@ namespace stan {
      */
     void generate_constructor(const program& prog,
                               const std::string& model_name, std::ostream& o) {
-      // constructor without RNG or template parameter
+      // constructor without seed or template parameter
       o << INDENT << model_name << "(stan::io::var_context& context__," << EOL;
       o << INDENT << "    std::ostream* pstream__ = 0)" << EOL;
       o << INDENT2 << ": prob_grad(0) {" << EOL;
-      o << INDENT2 << "typedef boost::ecuyer1988 rng_t;" << EOL;
-      o << INDENT2 << "rng_t base_rng(0);  // 0 seed default" << EOL;
-      o << INDENT2 << "ctor_body(context__, base_rng, pstream__);" << EOL;
+      o << INDENT2 << "ctor_body(context__, 0, pstream__);" << EOL;
       o << INDENT << "}" << EOL2;
-      // constructor with specified RNG
-      o << INDENT << "template <class RNG>" << EOL;
+      // constructor with specified seed
       o << INDENT << model_name << "(stan::io::var_context& context__," << EOL;
-      o << INDENT << "    RNG& base_rng__," << EOL;
+      o << INDENT << "    unsigned int random_seed__," << EOL;
       o << INDENT << "    std::ostream* pstream__ = 0)" << EOL;
       o << INDENT2 << ": prob_grad(0) {" << EOL;
-      o << INDENT2 << "ctor_body(context__, base_rng__, pstream__);" << EOL;
+      o << INDENT2 << "ctor_body(context__, random_seed__, pstream__);" << EOL;
       o << INDENT << "}" << EOL2;
       // body of constructor now in function
-      o << INDENT << "template <class RNG>" << EOL;
       o << INDENT << "void ctor_body(stan::io::var_context& context__," << EOL;
-      o << INDENT << "               RNG& base_rng__," << EOL;
+      o << INDENT << "               unsigned int random_seed__," << EOL;
       o << INDENT << "               std::ostream* pstream__) {" << EOL;
+      o << INDENT2 << "boost::ecuyer1988 base_rng__ =" << EOL;
+      o << INDENT2 << "  stan::services::util::create_rng(random_seed__, 0);"
+        << EOL;
+      o << INDENT2 << "(void) base_rng__;  // suppress unused var warning"
```

Comment:
Transformed data can have `_rng` functions.  I thought I'd already replied that, but I don't see it now.